# Desempenho de vereadores e vereadoras do Partido Socialismo e Liberdade (PSOL) na cidade do Rio de Janeiro durante as eleições municipais de 2020

O objetivo dessa consulta é verificar a distribuição de votos para os cargos de Vereador do município do Rio de Janeiro no ano de 2020 a partir de um olhar regional. Nesse projeto, utilizo os micro dados disponibilizados pelo Tribunal Superior Eleitoral (https://dadosabertos.tse.jus.br/dataset/) para ter acesso aos dados depositados em cada Seção Eleitoral da cidade. Para poder realizar a consulta aos bairros e endereços que ficam localizadas as Zona Eleitoral, acessei o banco de dados disponibilizado pelo Tribunal Regional do Estado do Rio de Janeiro (TRE-RJ), disponível aqui (https://www.tre-rj.jus.br/o-tre/enderecos-e-telefones/lista-completa-de-enderecos-telefones-e-nome-dos-juizes-dos-cartorios-zonas-eleitorais-com-e-mail-para-contato).

Responsável: João Leonardo /
Contato: joao.leonard@outlook.com

# Importando bibliotecas e interface gráfica

In [1]:
# Importando bibliotecas

import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [2]:
# Importanto interface gráfica

%pip install -U plotly
%pip install nbconvert

Note: you may need to restart the kernel to use updated packages.
Note: you may need to restart the kernel to use updated packages.


In [3]:
# Alterando o formato de exibição dos gráficos do Pandas

pd.options.plotting.backend = 'plotly'

# Limpando e organizando os bancos de dados disponíveis

In [4]:
# Lendo meus arquivos CSV com pandas, transformando em DataFrames

df_votos = pd.read_csv(r"C:\Users\João Leonardo\Documents\votacao_secao_2020_RJ.csv", sep = ';', encoding = 'iso-8859-1')
df_enderecos = pd.read_csv(r"C:\Users\João Leonardo\Documents\csv.csv", sep = ',')

In [5]:
# Visualizando as colunas do banco de dados eleitorais das eleições de 2020

df_votos.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3585762 entries, 0 to 3585761
Data columns (total 23 columns):
 #   Column            Dtype 
---  ------            ----- 
 0   DT_GERACAO        object
 1   HH_GERACAO        object
 2   ANO_ELEICAO       int64 
 3   CD_TIPO_ELEICAO   int64 
 4   NM_TIPO_ELEICAO   object
 5   NR_TURNO          int64 
 6   CD_ELEICAO        int64 
 7   DS_ELEICAO        object
 8   DT_ELEICAO        object
 9   TP_ABRANGENCIA    object
 10  SG_UF             object
 11  SG_UE             int64 
 12  NM_UE             object
 13  CD_MUNICIPIO      int64 
 14  NM_MUNICIPIO      object
 15  NR_ZONA           int64 
 16  NR_SECAO          int64 
 17  CD_CARGO          int64 
 18  DS_CARGO          object
 19  NR_VOTAVEL        int64 
 20  NM_VOTAVEL        object
 21  QT_VOTOS          int64 
 22  NR_LOCAL_VOTACAO  int64 
dtypes: int64(12), object(11)
memory usage: 629.2+ MB


In [6]:
df_votos_drop = df_votos.drop(['DT_GERACAO', 'HH_GERACAO', 'NM_TIPO_ELEICAO', 'TP_ABRANGENCIA', 'NR_SECAO', 'SG_UE', 'NR_LOCAL_VOTACAO'], axis = 1)
df_votos_rio_2020 = df_votos_drop[df_votos_drop['CD_MUNICIPIO'] == 60011]
df_votos_rio_2020

ANO_ELEICAO  CD_TIPO_ELEICAO  NR_TURNO  CD_ELEICAO  \
197286          2020                2         1         426   
197287          2020                2         1         426   
197288          2020                2         1         426   
197289          2020                2         1         426   
197290          2020                2         1         426   
...              ...              ...       ...         ...   
3585757         2020                2         1         426   
3585758         2020                2         1         426   
3585759         2020                2         1         426   
3585760         2020                2         1         426   
3585761         2020                2         1         426   

                       DS_ELEICAO  DT_ELEICAO SG_UF           NM_UE  \
197286   ELEIÇÕES MUNICIPAIS 2020  15/11/2020    RJ  RIO DE JANEIRO   
197287   ELEIÇÕES MUNICIPAIS 2020  15/11/2020    RJ  RIO DE JANEIRO   
197288   ELEIÇÕES MUNICIPAIS 2020  15/11/2020    RJ  RIO DE JANEIRO   
197289   ELEIÇÕES MUNICIPAIS 2020  15/11/2020    RJ  RIO DE JANEIRO   
197290   ELEIÇÕES MUNICIPAIS 2020  15/11/2020    RJ  RIO DE JANEIRO   
...                           ...         ...   ...             ...   
3585757  ELEIÇÕES MUNICIPAIS 2020  15/11/2020    RJ  RIO DE JANEIRO   
3585758  ELEIÇÕES MUNICIPAIS 2020  15/11/2020    RJ  RIO DE JANEIRO   
3585759  ELEIÇÕES MUNICIPAIS 2020  15/11/2020    RJ  RIO DE JANEIRO   
3585760  ELEIÇÕES MUNICIPAIS 2020  15/11/2020    RJ  RIO DE JANEIRO   
3585761  ELEIÇÕES MUNICIPAIS 2020  15/11/2020    RJ  RIO DE JANEIRO   

         CD_MUNICIPIO    NM_MUNICIPIO  NR_ZONA  CD_CARGO  DS_CARGO  \
197286          60011  RIO DE JANEIRO      218        11  PREFEITO   
197287          60011  RIO DE JANEIRO      218        11  PREFEITO   
197288          60011  RIO DE JANEIRO      218        11  PREFEITO   
197289          60011  RIO DE JANEIRO      218        11  PREFEITO   
197290          60011  RIO DE JANEIRO      218        11  PREFEITO   
...               ...             ...      ...       ...       ...   
3585757         60011  RIO DE JANEIRO      185        13  VEREADOR   
3585758         60011  RIO DE JANEIRO      185        13  VEREADOR   
3585759         60011  RIO DE JANEIRO      185        13  VEREADOR   
3585760         60011  RIO DE JANEIRO      185        13  VEREADOR   
3585761         60011  RIO DE JANEIRO      185        13  VEREADOR   

         NR_VOTAVEL                                    NM_VOTAVEL  QT_VOTOS  
197286           15                          PAULO SANTOS MESSINA        16  
197287           10                      MARCELO BEZERRA CRIVELLA        58  
197288           12                      MARTHA MESQUITA DA ROCHA        22  
197289           13                       BENEDITA SOUZA DA SILVA        27  
197290           17  LUIZ EDUARDO CARNEIRO DA SILVA DE SOUZA LIMA        12  
...             ...                                           ...       ...  
3585757          25                                    Democratas         2  
3585758          13                     Partido dos Trabalhadores         2  
3585759          12               Partido Democrático Trabalhista         1  
3585760          11                                 PROGRESSISTAS         1  
3585761          10                                  REPUBLICANOS         2  

[1421785 rows x 16 columns]

In [7]:
# Visualizando o banco de dados com os endereços das Zonas Eleitorais

df_enderecos.sample(3)

ZE                            ENDEREÇO           BAIRRO          CIDADE  \
24  170  AV. PROFESSOR MANOEL DE ABREU, 286         MARACANÃ  RIO DE JANEIRO   
34  211           RUA JARDIM BOTÂNICO, 1060  JARDIM BOTÂNICO  RIO DE JANEIRO   
7    16         RUA CONDE DE BAEPENDI, N 40      LARANJEIRAS  RIO DE JANEIRO   

                       TELEFONES                E-MAIL  \
24  (21) 2284-5504(21) 2254-1453  zon170@tre-rj.jus.br   
34  (21) 2249-7610(21) 2259-6534  zon211@tre-rj.jus.br   
7                 (21) 2265-5197  zon016@tre-rj.jus.br   

                           JUIZ        MAPA  
24     SANDRO PITTHAN ESPINDOLA  Visualizar  
34        CARLOS AUGUSTO BORGES  Visualizar  
7   MARCEL LAGUNA DUQUE ESTRADA  Visualizar

In [8]:
# Excluindo colunas desnecessárias para análise

df_enderecos_zonas = df_enderecos.reindex(columns=['ZE', 'ENDEREÇO', 'BAIRRO']) \
                                    .rename(columns={'ZE':'NR_ZONA'})
df_enderecos_zonas.head()

NR_ZONA                         ENDEREÇO           BAIRRO
0        4        RUA JARDIM BOTÂNICO, 1060  JARDIM BOTÂNICO
1        5             RUA MIGUEL LEMOS, 97       COPACABANA
2        7          RUA ANTONIO BASILIO, 76           TIJUCA
3        8  AVENIDA DOM HÉLDER CÂMARA, 4175     DEL CASTILHO
4        9     AV AYRTON SENNA 2001 BLOCO C  BARRA DA TIJUCA

In [9]:
# Unindo os bancos de dados

df_votos_bairro = pd.merge(df_votos_rio_2020, df_enderecos_zonas, how = 'left')
df_votos_bairro.head()

ANO_ELEICAO  CD_TIPO_ELEICAO  NR_TURNO  CD_ELEICAO  \
0         2020                2         1         426   
1         2020                2         1         426   
2         2020                2         1         426   
3         2020                2         1         426   
4         2020                2         1         426   

                 DS_ELEICAO  DT_ELEICAO SG_UF           NM_UE  CD_MUNICIPIO  \
0  ELEIÇÕES MUNICIPAIS 2020  15/11/2020    RJ  RIO DE JANEIRO         60011   
1  ELEIÇÕES MUNICIPAIS 2020  15/11/2020    RJ  RIO DE JANEIRO         60011   
2  ELEIÇÕES MUNICIPAIS 2020  15/11/2020    RJ  RIO DE JANEIRO         60011   
3  ELEIÇÕES MUNICIPAIS 2020  15/11/2020    RJ  RIO DE JANEIRO         60011   
4  ELEIÇÕES MUNICIPAIS 2020  15/11/2020    RJ  RIO DE JANEIRO         60011   

     NM_MUNICIPIO  NR_ZONA  CD_CARGO  DS_CARGO  NR_VOTAVEL  \
0  RIO DE JANEIRO      218        11  PREFEITO          15   
1  RIO DE JANEIRO      218        11  PREFEITO          10   
2  RIO DE JANEIRO      218        11  PREFEITO          12   
3  RIO DE JANEIRO      218        11  PREFEITO          13   
4  RIO DE JANEIRO      218        11  PREFEITO          17   

                                     NM_VOTAVEL  QT_VOTOS  \
0                          PAULO SANTOS MESSINA        16   
1                      MARCELO BEZERRA CRIVELLA        58   
2                      MARTHA MESQUITA DA ROCHA        22   
3                       BENEDITA SOUZA DA SILVA        27   
4  LUIZ EDUARDO CARNEIRO DA SILVA DE SOUZA LIMA        12   

                            ENDEREÇO     BAIRRO  
0  RUA SIDÔNIO PAES, 54, LOJAS 4 E 5  CASCADURA  
1  RUA SIDÔNIO PAES, 54, LOJAS 4 E 5  CASCADURA  
2  RUA SIDÔNIO PAES, 54, LOJAS 4 E 5  CASCADURA  
3  RUA SIDÔNIO PAES, 54, LOJAS 4 E 5  CASCADURA  
4  RUA SIDÔNIO PAES, 54, LOJAS 4 E 5  CASCADURA

# Distribuição da votação em candidatos e candidatas do PSOL organizada por bairro

Tarcísio Motta (50123)

In [10]:
votos_tarcisio_2020 = df_votos_bairro[df_votos_bairro['NR_VOTAVEL'] == 50123]
votos_tarcisio_2020_org = votos_tarcisio_2020.groupby(by='BAIRRO').sum() \
                    .reset_index() \
                    .sort_values(by=["QT_VOTOS"], ascending = False)
votos_tarcisio_2020_org.plot.bar(x = 'BAIRRO', y = 'QT_VOTOS')

In [11]:
# Soma do total de votos vereador Tarcisio Mota

total_votos_tarcisio = votos_tarcisio_2020_org['QT_VOTOS'].sum()
print("O vereador Tarcísio Mota obteve um total de", total_votos_tarcisio, "votos na cidade do Rio de Janeiro.")

O vereador Tarcísio Mota obteve um total de 86243 votos na cidade do Rio de Janeiro.


Chico Alencar (50500)

In [12]:
votos_chico_2020 = df_votos_bairro[df_votos_bairro['NR_VOTAVEL'] == 50500]
votos_chico_2020_org = votos_chico_2020.groupby(by='BAIRRO').sum() \
                    .reset_index() \
                    .sort_values(by=["QT_VOTOS"], ascending = False)
votos_chico_2020_org.plot.bar(x = 'BAIRRO', y = 'QT_VOTOS')

In [13]:
# Soma do total de votos vereador Chico Alencar

total_votos_chico = votos_chico_2020_org['QT_VOTOS'].sum()
print("O vereador Chico Alencar obteve um total de", total_votos_chico, "votos na cidade do Rio de Janeiro.")

O vereador Chico Alencar obteve um total de 49422 votos na cidade do Rio de Janeiro.


Monica Benício (50333)

In [14]:
votos_monica_2020 = df_votos_bairro[df_votos_bairro['NR_VOTAVEL'] == 50333]
votos_monica_2020_org = votos_monica_2020.groupby(by='BAIRRO').sum() \
                    .reset_index() \
                    .sort_values(by=["QT_VOTOS"], ascending = False)
votos_monica_2020_org.plot.bar(x = 'BAIRRO', y = 'QT_VOTOS')

In [15]:
# Soma do total de votos vereadora Mônica Benício

total_votos_monica = votos_monica_2020_org['QT_VOTOS'].sum()
print("A vereadora Monica Benício obteve um total de", total_votos_monica, "votos na cidade do Rio de Janeiro.")

A vereadora Monica Benício obteve um total de 22919 votos na cidade do Rio de Janeiro.


Paulo Pinheiro (50111)

In [16]:
votos_paulo_2020 = df_votos_bairro[df_votos_bairro['NR_VOTAVEL'] == 50111]
votos_paulo_2020_org = votos_paulo_2020.groupby(by='BAIRRO').sum() \
                    .reset_index() \
                    .sort_values(by=["QT_VOTOS"], ascending = False)
votos_paulo_2020_org.plot.bar(x = 'BAIRRO', y = 'QT_VOTOS')

In [17]:
# Soma do total de votos vereador Paulo Pinheiro

total_votos_paulo = votos_paulo_2020_org['QT_VOTOS'].sum()
print("O vereador Paulo Pinheiro obteve um total de", total_votos_paulo, "votos na cidade do Rio de Janeiro.")

O vereador Paulo Pinheiro obteve um total de 14760 votos na cidade do Rio de Janeiro.


Thaís Ferreira (50010)

In [18]:
votos_thais_2020 = df_votos_bairro[df_votos_bairro['NR_VOTAVEL'] == 50010]
votos_thais_2020_org = votos_thais_2020.groupby(by='BAIRRO').sum() \
                    .reset_index() \
                    .sort_values(by=["QT_VOTOS"], ascending = False)
votos_thais_2020_org.plot.bar(x = 'BAIRRO', y = 'QT_VOTOS')

In [19]:
# Soma do total de votos vereadora Thaís Ferreira

total_votos_thais = votos_thais_2020_org['QT_VOTOS'].sum()
print("A vereadora Thaís Ferreira obteve um total de", total_votos_thais, "votos na cidade do Rio de Janeiro.")

A vereadora Thaís Ferreira obteve um total de 14284 votos na cidade do Rio de Janeiro.


William Siri (50222)

In [20]:
votos_siri_2020 = df_votos_bairro[df_votos_bairro['NR_VOTAVEL'] == 50222]
votos_siri_2020_org = votos_siri_2020.groupby(by='BAIRRO').sum() \
                    .reset_index() \
                    .sort_values(by=["QT_VOTOS"], ascending = False)
votos_siri_2020_org.plot.bar(x = 'BAIRRO', y = 'QT_VOTOS')

In [21]:
# Soma do total de votos vereador William Siri

total_votos_siri = votos_siri_2020_org['QT_VOTOS'].sum()
print("O vereador William Siri obteve um total de", total_votos_siri, "votos na cidade do Rio de Janeiro.")

O vereador William Siri obteve um total de 9957 votos na cidade do Rio de Janeiro.


Dr. Marcos Paulo (50505)

In [22]:
votos_mpaulo_2020 = df_votos_bairro[df_votos_bairro['NR_VOTAVEL'] == 50505]
votos_mpaulo_2020_org = votos_mpaulo_2020.groupby(by='BAIRRO').sum() \
                    .reset_index() \
                    .sort_values(by=["QT_VOTOS"], ascending = False)
votos_mpaulo_2020_org.plot.bar(x = 'BAIRRO', y = 'QT_VOTOS')

In [23]:
# Soma do total de votos vereador Dr. Marcos Paulo

total_votos_mpaulo = votos_mpaulo_2020_org['QT_VOTOS'].sum()
print("O vereador Dr. Marcos Paulo obteve um total de", total_votos_mpaulo, "votos na cidade do Rio de Janeiro.")

O vereador Dr. Marcos Paulo obteve um total de 9009 votos na cidade do Rio de Janeiro.


# Considerações Finais

Nas eleições municipais de 2020 os candidatos e as candidatas eleitos e eleitas pelo Partido Socialismo e Liberdade (PSOL) receberam um total de 206594 votos do eleitorado carioca. Destes:

- 18% foram em canidatas do gênero feminino, enquanto 82% foram para candidatos do gênero masculino. 

- Das 7 candidaturas eleitas, 6 delas tem seus principais colégios eleitorais nos bairros do Jardim Botânico, Maracanã e Barra da Tijuca. O único vereador eleito que distoa é William Siri, tendo como principal reduto eleitoral o bairro de Campo Grande, o mais populoso da cidade.

- De um modo geral, os bairros de Marechal Hermes, Deodoro e Jardim Sulacap são os que as candidaturas obtiveram desempenhos mais tímidos, em relação às demais regiões da cidade.

---------

Essa é uma primeira tentativa de análise de dados eleitorais do PSOL, que tem por objetivo contribuir para tomada de decisões intra partidárias em eleições subsequentes.

Caso tenha algum comentário, dúvida ou pergunta a ser feita sobre esse conjunto de dados e a análise aqui feita, é só entrar em contato comigo.

Até a próxima!